### ISSUE TITLE
Prevent `batch_size` from influencing the number of training samples on TensorflowTrainer

### ISSUE THREAD

**Issue by kevinmartinjos** [capreolus-ir/capreolus#76]
PyTorch does `batches_per_epoch = self.config["itersize"] // self.config["batch"]`
TensorflowTrainer follows a different convention: each "epoch" consists of "itersize" iterations where each iteration has "batch_size" batches

It won't  make much sense to tie batch_size to the number of examples in an iteration. There's no need for one to be dependent on the other. This would mean that if you move to a machine with less GPU memory, both the number of steps and the batch size would have to be adjusted to achieve a similar training setup. (The problem is similar to how the LR becomes dependent on the batch size if you sum the loss rather than averaging it.)
